x/nft/internal/keeper/querier.go

```diff
@@ -0,0 +1,143 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/nft/internal/types"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// query endpoints supported by the NFT Querier
+const (
+	QuerySupply       = "supply"
+	QueryOwner        = "owner"
+	QueryOwnerByDenom = "ownerByDenom"
+	QueryCollection   = "collection"
+	QueryDenoms       = "denoms"
+	QueryNFT          = "nft"
+)
+
+// NewQuerier is the module level router for state queries
+func NewQuerier(k Keeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err sdk.Error) {
+		switch path[0] {
+		case QuerySupply:
+			return querySupply(ctx, path[1:], req, k)
+		case QueryOwner:
+			return queryOwner(ctx, path[1:], req, k)
+		case QueryOwnerByDenom:
+			return queryOwnerByDenom(ctx, path[1:], req, k)
+		case QueryCollection:
+			return queryCollection(ctx, path[1:], req, k)
+		case QueryNFT:
+			return queryNFT(ctx, path[1:], req, k)
+		case QueryDenoms:
+			return queryDenoms(ctx, path[1:], req, k)
+		default:
+			return nil, sdk.ErrUnknownRequest("unknown nft query endpoint")
+		}
+	}
+}
+
+func queryDenoms(ctx sdk.Context, path []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
+	denoms := k.GetDenoms(ctx)
+	bz := types.ModuleCdc.MustMarshalJSON(denoms)
+	return bz, nil
+}
+
+func querySupply(ctx sdk.Context, path []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
```

Comment:
should we also create a general supply querier to return the supply for all the Collections of the chain?

Reply:
maybe we just make denom optional in the query so that if it's missing it returns the supply for all collections?